metrics/prometheus.go

```diff
@@ -384,12 +409,28 @@ func (p *Prometheus) MeasureServe(routeID, host, method string, code int, start
 	method = measuredMethod(method)
 	t := p.sinceS(start)
 
-	if p.opts.EnableServeRouteMetrics {
-		p.serveRouteM.WithLabelValues(fmt.Sprintf("%d", code), method, routeID).Observe(t)
+	if p.opts.EnableServeRouteMetrics || p.opts.EnableServeHostMetrics {
```

Comment:
I think this check may be avoided by utilizing nil slice trick like
```diff
-       if p.opts.EnableServeRouteMetrics || p.opts.EnableServeHostMetrics {
-               metrics := []string{}
-               if p.opts.EnableServeStatusCodeMetric {
-                       metrics = append(metrics, fmt.Sprint(code))
-               }
-               if p.opts.EnableServeMethodMetric {
-                       metrics = append(metrics, method)
-               }
-               if p.opts.EnableServeRouteMetrics {
-                       p.serveRouteM.WithLabelValues(append(metrics, routeID)...).Observe(t)
-               }
-               if p.opts.EnableServeHostMetrics {
-                       p.serveHostM.WithLabelValues(append(metrics, hostForKey(host))...).Observe(t)
-               }
+       var metrics []string
+       if p.opts.EnableServeStatusCodeMetric {
+               metrics = append(metrics, fmt.Sprint(code))
+       }
+       if p.opts.EnableServeMethodMetric {
+               metrics = append(metrics, method)
+       }
+
+       if p.opts.EnableServeRouteMetrics {
+               p.serveRouteM.WithLabelValues(append(metrics, routeID)...).Observe(t)
+       }
+       if p.opts.EnableServeHostMetrics {
+               p.serveHostM.WithLabelValues(append(metrics, hostForKey(host))...).Observe(t)
        }
```

Reply:
if we do that it will be the situation it was before. [You mentioned](https://github.com/zalando/skipper/pull/1755#discussion_r614323607) that it would "be executed even when `EnableServeRouteMetrics` or/and `EnableServeHostMetrics` are disabled."

Reply:
Right, we definitely have a conflict between readability and peanut savings, will resolve :)